nuxeo-drive-client/nxdrive/engine/engine.py

```diff
@@ -979,20 +980,38 @@ def get_user_full_name(self, userid):
         """
             Get the last contributor full name
         """
-        fullname = userid
         try:
-            if userid in self._user_cache:
-                fullname = self._user_cache[userid]
-            else:
-                rest_client = self.get_rest_api_client()
-                response = rest_client.get_user_full_name(userid)
-                if response and 'properties' in response:
-                    properties = response['properties']
-                    firstName = properties.get('firstName')
-                    lastName = properties.get('lastName')
-                    if firstName and lastName:
-                        fullname = " ".join([firstName, lastName]).strip()
-                        self._user_cache[userid] = fullname
+            rest_client = self.get_rest_api_client()
+            response = rest_client.get_user_full_name(userid)
+            if response and 'properties' in response:
+                properties = response['properties']
+                firstName = properties.get('firstName')
+                lastName = properties.get('lastName')
+                if self._dao.get_user_info(userid):
+                    self._dao.update_user_info(userid, firstName, lastName)
+                else:
+                    self._dao.insert_user_info(userid, firstName, lastName)
         except urllib2.URLError as e:
             log.exception(e)
-        return fullname
+        except sqlite3.OperationalError as e:
```

Comment:
I don't think those exceptions should leek in here, but be contained in their respective functions, e.g. url error in the get_user_fullname method which would return None if it fails and the caller already checks for a valid response.
Similarly for the database errors.